### Forbe.com
> C'est vraiment une belle introduction, bravo Nico!

### Mark Zuckerberg
> Best project ever!! 9.5/10




In [1]:
#essential imports
import pandas as pd
import numpy as np
import json
from pandas.io.json import json_normalize

#scraping imports
import requests
from bs4 import BeautifulSoup

#plotting imports
%matplotlib inline
import matplotlib.pyplot as plt

#String matching
import re

#date
import datetime as dt

# Amazon Dataset

Amazon has a lot of categories:

Books, Electronics, Movies and TV, CDs and Vinyl, Clothing (Shoes and Jewelry), Home and Kitchen, Kindle Store, Sports and Outdoors, Cell Phones and Accessorie, Health and Personal Care, Toys and Games, Video Games, Tools and Home Improvement, Beauty, Apps for Android, Office Products, Pet Supplies, Automotive, Grocery and Gourmet Food, Patio (Lawn and Garden), Baby, Digital Music, Musical Instruments, Amazon Instant Video


### For our project the useful categories are:
* Books
* Movies and TV
* CDs and Vinyl
* Kindle Store
* Digital Music
* Amazon Instant Video

#### Potentially useful
* Toy and Games
* Video Games

---
### [Amazon DB](http://jmcauley.ucsd.edu/data/amazon/links.html)

### Authors
* [Books](http://snap.stanford.edu/data/amazon/productGraph/categoryFiles/reviews_Books_5.json.gz)
* [Kindle Store](http://snap.stanford.edu/data/amazon/productGraph/categoryFiles/reviews_Kindle_Store_5.json.gz)

### Actors
* [Movies and TV](http://snap.stanford.edu/data/amazon/productGraph/categoryFiles/reviews_Movies_and_TV_5.json.gz)
* [Amazon Instant Video](http://snap.stanford.edu/data/amazon/productGraph/categoryFiles/reviews_Amazon_Instant_Video_5.json.gz)

### Musician
* [CDs and Vinyl](http://snap.stanford.edu/data/amazon/productGraph/categoryFiles/reviews_CDs_and_Vinyl_5.json.gz)
* [Digital Music](http://snap.stanford.edu/data/amazon/productGraph/categoryFiles/reviews_Digital_Music_5.json.gz)

---

In [2]:
data_path = 'data/'

In [3]:
#data = pd.read_json(data_path+'Kindle_Store_5.json', lines=True)

In [4]:
#data

asin   helpful  overall  \
0       B000F83SZQ    [0, 0]        5   
1       B000F83SZQ    [2, 2]        4   
2       B000F83SZQ    [2, 2]        4   
3       B000F83SZQ    [1, 1]        5   
4       B000F83SZQ    [0, 1]        4   
5       B000F83SZQ    [0, 0]        4   
6       B000F83SZQ    [0, 0]        4   
7       B000F83SZQ    [1, 1]        4   
8       B000FA64PA    [0, 0]        5   
9       B000FA64PA    [0, 0]        4   
10      B000FA64PA    [0, 0]        5   
11      B000FA64PA    [0, 0]        4   
12      B000FA64PA    [0, 0]        3   
13      B000FA64PK    [0, 0]        3   
14      B000FA64PK    [0, 0]        5   
15      B000FA64PK    [0, 0]        5   
16      B000FA64PK    [0, 0]        5   
17      B000FA64PK    [4, 4]        3   
18      B000FA64PK    [0, 1]        5   
19      B000FA64PK    [0, 0]        4   
20      B000FA64PK    [0, 0]        5   
21      B000FA64QO    [0, 0]        2   
22      B000FA64QO    [0, 0]        5   
23      B000FA64QO    [2, 2]        4   
24      B000FA64QO    [0, 0]        5   
25      B000FA64QO    [0, 0]        3   
26      B000FBFMVG    [0, 0]        5   
27      B000FBFMVG    [5, 5]        3   
28      B000FBFMVG    [0, 1]        4   
29      B000FBFMVG    [0, 0]        5   
...            ...       ...      ...   
982589  B00M0RE7CS    [1, 2]        5   
982590  B00M0RE7CS    [1, 1]        5   
982591  B00M0RE7CS    [1, 1]        5   
982592  B00M0RE7CS  [11, 11]        5   
982593  B00M0RE7CS    [9, 9]        5   
982594  B00M0RE7CS    [1, 2]        5   
982595  B00M0RE7CS    [5, 5]        5   
982596  B00M0RE7CS    [1, 2]        5   
982597  B00M0RE7CS    [2, 2]        5   
982598  B00M0RE7CS    [1, 1]        5   
982599  B00M0RE7CS    [1, 2]        5   
982600  B00M0RE7CS    [4, 4]        5   
982601  B00M0RE7CS    [7, 7]        5   
982602  B00M0RE7CS    [7, 7]        5   
982603  B00M0RE7CS    [1, 2]        5   
982604  B00M0RE7CS    [1, 1]        5   
982605  B00M0RE7CS    [2, 2]        5   
982606  B00M0RE7CS    [3, 3]        5   
982607  B00M0RE7CS    [5, 5]        5   
982608  B00M0RE7CS    [1, 2]        5   
982609  B00M0RE7CS    [1, 1]        5   
982610  B00M0RE7CS    [1, 1]        5   
982611  B00M0RE7CS    [3, 3]        5   
982612  B00M13FNSS    [2, 2]        4   
982613  B00M13FNSS    [2, 2]        5   
982614  B00M13FNSS    [2, 2]        5   
982615  B00M13FNSS    [0, 0]        5   
982616  B00M13FNSS    [1, 1]        5   
982617  B00M13FNSS    [0, 0]        5   
982618  B00M13FNSS    [2, 2]        5   

                                               reviewText   reviewTime  \
0       I enjoy vintage books and movies so I enjoyed ...   05 5, 2014   
1       This book is a reissue of an old one; the auth...   01 6, 2014   
2       This was a fairly interesting read.  It had ol...   04 4, 2014   
3       I'd never read any of the Amy Brewster mysteri...  02 19, 2014   
4       If you like period pieces - clothing, lingo, y...  03 19, 2014   
5       A beautiful in-depth character description mak...  05 26, 2014   
6       I enjoyed this one tho I'm not sure why it's c...  06 10, 2014   
7       Never heard of Amy Brewster. But I don't need ...  03 22, 2014   
8       Darth Maul working under cloak of darkness com...  10 11, 2013   
9       This is a short story focused on Darth Maul's ...  02 13, 2011   
10      I think I have this one in both book and audio...  01 27, 2014   
11      Title has nothing to do with the story.  I did...  09 17, 2011   
12      Well written. Interesting to see Sideous (thro...  12 31, 2013   
13      Troy Denning's novella Recovery was originally...  03 15, 2012   
14      I am not for sure on how much of a difference ...  05 12, 2013   
15      I really enjoyed the book. Had the normal back...   01 2, 2014   
16      Great read enjoyed every minute of it .  I thi...  10 29, 2013   
17      Another well written eBook by Troy Denning, bu...  04 16, 2009   
18      This one promises to be another good book. I h...  01 27,

# Scraping Wikipedia

We want to scrap all the musicians, actors and authors' dead on wikipedia. Since we want the most notorious ones, we will scrap from [the page summarizing the year](https://en.wikipedia.org/w/index.php?title=2000#Deaths), and not the specific [page for the deaths](https://en.wikipedia.org/wiki/Deaths_in_2000).

We will iterate for each wanted year page, and scrap the celebrity's name, birth date, description, and death date.

In [5]:
# Function that matchs a line of the wikitext and return an array of tuples: 
# (person_death_date, person_name, person_description, person_birth_date)
# debug = True return the intermediary matching
def matchLine(line, debug=False):
    if debug:
        print(line)
    #match when a line contains only 1 celebrity
    match_1_line = re.match( r'.*\[\[(.*?)\]\] (&ndash;|-|–).*?\[\[(.*?)\]\](.*)\(b\..*\[\[(.*?)\]\]\)', line)
    if match_1_line:
        if debug:
            print(match_1_line)
        return [(match_1_line.group(1),match_1_line.group(3),match_1_line.group(4),match_1_line.group(5))]
    #didn't find a match for 1 celebrity
    else:
        result = []
        #consider it's a line with multiple celebrities dead on the same day, separated by \n**
        s = line.split("\n**")
        #if the split didn't work, we return a matching error:
        if len(s)==1:
            print("No match found for: "+ str(year) +" "+ line)
        #the first split contains only the death date
        match_date = re.match( r'\[\[(.*)\]\]',s[0]).group(1)
        #iterate for each celebrities
        for i in range(1,len(s)):
            match_3_param = re.match( r'.*\[\[(.*?)\]\](.*)\(b\. ?\[\[(.*?)\]\]\)', s[i])
            if debug:
                print(match_3_param)
            #if the match is succesful, add the celebrity to the array
            if match_3_param and match_date:
                result.append((match_date,match_3_param.group(1),match_3_param.group(2),match_3_param.group(3)))
            #otherwise return a matching error
            else:
                print("No match found for: "+ str(year) +" "+ s[i])
        return result

In [6]:
month_list = ['January', 'February', 'March', 'April', 'May', 
              'June', 'July', 'August', 'September', 'October', 'November', 'December']

# Function that return a standardized date in string %year-%month-%day
def computeDate(year, md):
    a = md.split(" ")
    if a[0] in month_list:
        month = month_list.index(a[0])+1
        day = a[1]
        return str(year)+"-"+str(month).zfill(2)+"-"+str(day).zfill(2)
    #return an date error that we couldn't convert
    else:
        print("Date error: "+md+" "+str(year));
        return np.nan

In [7]:
#create an empty dataFrame
columns = ['Death Date', 'Name', 'Description','Birth Date']
df = pd.DataFrame(columns=columns) 

#the year of the interval of the amazon dataset
starting_year = 1996
ending_year = 2014

#every year we scrap the corresponding wikipedia page
for year in range(starting_year,ending_year+1):
    #we retrieve only the wikitext using the parameter action=raw
    #note that the api documentation advise to do this way if we are interested only by the wikitext
    year_url = "https://en.wikipedia.org/w/index.php?action=raw&title={}&maxlag=5".format(year)
    r = requests.get(year_url)
    page_text = r.text
    #we split the text to retrieve only the part about celebrities' deaths
    if("== Deaths ==" in page_text):
        a = page_text.split("== Deaths ==")[1]
    elif("==Deaths==" in page_text):
        a = page_text.split("==Deaths==")[1]
    else:
        a = None
    #we set our starting text at the first month (i.e January)
    if("=== January ===" in page_text):
        a = a.split("=== January ===")[1]
    elif("===January===" in page_text):
        a = a.split("===January===")[1]
    else:
        a = None
    #all the month are separated by \n\n, so we split and iterate over the 12
    deaths = a.split("\n\n")
    for i in range(12):
        #We have 2 case: 1 celebrity per date, or multiple celebrity per date, so we split in consequence,
        #then we match the line into groups using pattern recognition.
        s = deaths[i].split('* ',1)[1]
        lines = s.split("\n* ")
        for line in lines:
            res = matchLine(line)
            for person in res:
                    #when we finished matching, we format the date, then add the celebrity into the dataFrame
                    date = computeDate(year,person[0])
                    df.loc[len(df)]=[date,person[1]," "+person[2].lower(),person[3]]
print(df.shape)
df.head(10)

No match found for: 1996  Victims of [[TWA Flight 800]]
No match found for: 2001  2,996 people (2,977 victims and 19 hijackers) who died in the [[September 11 attacks]]
No match found for: 2013 [[April 30]] (death announced on this date) &ndash; [[Deanna Durbin]], Canadian-born singer and actress (b. [[1921]])
(3313, 4)


Death Date                                  Name  \
0  1996-01-01  Malladihalli Sri Raghavendra Swamiji   
1  1996-01-01                    Moshe Aryeh Freund   
2  1996-01-01                         Arleigh Burke   
3  1996-01-01                        Arthur Rudolph   
4  1996-01-02                        Karl Targownik   
5  1996-01-05                          Yahya Ayyash   
6  1996-01-05                      Lincoln Kirstein   
7  1996-01-05                      Richard Versalle   
8  1996-01-07             Prime Minister of Hungary   
9  1996-01-07                          Tarō Okamoto   

                                         Description Birth Date  
0                                     , indian yogi        1890  
1                                   , israeli rabbi        1894  
2                          , american naval officer        1901  
3                          , german rocket engineer        1906  
4   , hungarian psychiatrist and holocaust survivor        1915  
5                             , palestinian shaheed        1966  
6                  , american writer and impresario        1907  
7                         , american operatic tenor        1932  
8                                                          1930  
9                                 , japanese artist        1911

The matcher has only 3 errors over 3313 matching, which is pretty nice.
Most of those error is just a special change of format in the list, where they listed celebrities inside specific tragedies.
In this case we will add them manually.

In [8]:
#we add only wanted celebrity (singer, actor,...)
#TWA Flight 8000
df.loc[len(df)]=["1996-07-17","Marcel Dadi",", French guitarist".lower(),"1951"]
df.loc[len(df)]=["1996-07-17","David Hogan",", American composer".lower(),"1949"]
#Deanna Durbin
df.loc[len(df)]=["2013-04-30","Deanna Durbin",", Canadian-born singer and actress".lower(),"1921"]

We consider to multi-index on name and birth date.
Then we order them in fuction of their death date.

In [9]:
df2 = df.set_index(['Name', 'Birth Date'])
print("Index is unique: "+str(df.index.is_unique))
df2.sort_values('Death Date',inplace = True)
df2.head()

Index is unique: True


Death Date  \
Name                                 Birth Date               
Malladihalli Sri Raghavendra Swamiji 1890        1996-01-01   
Moshe Aryeh Freund                   1894        1996-01-01   
Arleigh Burke                        1901        1996-01-01   
Arthur Rudolph                       1906        1996-01-01   
Karl Targownik                       1915        1996-01-02   

                                                                                       Description  
Name                                 Birth Date                                                     
Malladihalli Sri Raghavendra Swamiji 1890                                           , indian yogi   
Moshe Aryeh Freund                   1894                                         , israeli rabbi   
Arleigh Burke                        1901                                , american naval officer   
Arthur Rudolph                       1906                                , german rocket engineer   
Karl Targownik                       1915         , hungarian psychiatrist and holocaust survivor

Then we removed the celebrities which died outside of the interval of the Amazon dataset (May 1996 - July 2014).

In [13]:
df2 = df2[(df2['Death Date'] >= '1996-05-01') & (df2['Death Date'] <= '2014-07-31')]
df2.tail(5)

Death Date  \
Name                 Birth Date               
David Easton         1917        2014-07-19   
James Garner         1928        2014-07-19   
Carlo Bergonzi       1924        2014-07-25   
Francesco Marchisano 1929        2014-07-27   
Julio Grondona       1931        2014-07-30   

                                                               Description  
Name                 Birth Date                                             
David Easton         1917         , canadian-american political scientist   
James Garner         1928                                , american actor   
Carlo Bergonzi       1924                       , italian tenor and actor   
Francesco Marchisano 1929                              , italian cardinal   
Julio Grondona       1931                , argentinian football authority

Now we want to keep only the celebrities **that are musician, actor or author.**
Since we have a small description of the celebrity, it's trivial that the description will contains his job if the celebrity is famous by his job. 
_Note that some celebrities doesn't have a description, it's the case when their name contains the description: president, prince, king,...etc._

We match a celebrity if he is a musician, actor or author if it contains a specific keyword, for example: _"actor"_.

In [11]:
df3 = df2

#return true if the description contains one of this keywork
jobMusician = ["dj","baritone","bard","pianist","singer","tenor ","soprano", "composer","trumpeter","saxophonist","lyricist", "drummer", "musician", "rapper","guitarist","violinist","violist","bassist"]
def isMusician(s):
    for job in jobMusician:
        if job in s:
            return True
    return False

#return true if the description contains one of this keywork
jobActor = ["actor", "actress","filmmaker","cinematographer","film director", "film producer"]
def isActor(s):
    for job in jobActor:
        if job in s:
            return True
    return False

#return true if the description contains one of this keywork
jobAuthor = ["autor", "author", "writer", "poet", "novelist","cartoonist","comic strip artist","manga artist"]
def isAuthor(s):
    for job in jobAuthor:
        if job in s:
            return True
    return False

#add 3 column to the data frame: Actor, author and Musician. It's possible that someone is both, so they all have those 3 booleans parameters.
df3 = df3.merge(df3.Description.apply(lambda s: pd.Series({'Musician':isMusician(s), 'Actor':isActor(s), 'Author':isAuthor(s)})), 
    left_index=True, right_index=True) 
df_artists = df3[(df3['Actor'] == True) | (df3['Author'] == True) | (df3['Musician'] == True)]
df_artists = df_artists.sort_values('Death Date')

print("Number of celebrities: "+str(df.shape[0]))
print("Number of useful celebrities: "+str(df_artists.shape[0]))
print("Number of unwanted celebrities: "+str(df.shape[0]-df_artists.shape[0]))

#print only the wanted celebrities
df_artists.head(5)

Number of celebrities: 3316
Number of useful celebrities: 1316
Number of unwanted celebrities: 2000


Death Date                             Description  \
Name          Birth Date                                                       
Jack Weston   1924        1996-05-03                       , american actor    
John Beradino 1917        1996-05-19   , american baseball player and actor    
Jon Pertwee   1919        1996-05-20                        , british actor    
Lash LaRue    1917        1996-05-21                       , american actor    
Paul Delph    1957        1996-05-21       , american musician and producer    

                          Actor  Author  Musician  
Name          Birth Date                           
Jack Weston   1924         True   False     False  
John Beradino 1917         True   False     False  
Jon Pertwee   1919         True   False     False  
Lash LaRue    1917         True   False     False  
Paul Delph    1957        False   False      True

Now we want to link a celebrity to all its work.

First let's start with the actors.

In [12]:
df_actors = df_artists[(df_artists['Actor'] == True)]
print(df_actors.shape)
df_actors.head()

(640, 5)


Death Date  \
Name                  Birth Date               
Jack Weston           1924        1996-05-03   
John Beradino         1917        1996-05-19   
Jon Pertwee           1919        1996-05-20   
Lash LaRue            1917        1996-05-21   
Enrique Álvarez Félix 1934        1996-05-24   

                                                             Description  \
Name                  Birth Date                                           
Jack Weston           1924                             , american actor    
John Beradino         1917         , american baseball player and actor    
Jon Pertwee           1919                              , british actor    
Lash LaRue            1917                             , american actor    
Enrique Álvarez Félix 1934                              , mexican actor    

                                  Actor  Author  Musician  
Name                  Birth Date                           
Jack Weston           1924         True   False     False  
John Beradino         1917         True   False     False  
Jon Pertwee           1919         True   False     False  
Lash LaRue            1917         True   False     False  
Enrique Álvarez Félix 1934         True   False     False

The idea is to go to the page of the actor, find the section _"Filmography"_ and read the corresponding tables in its subsections.

In [57]:
table = """{| class="wikitable sortable"
|-
! Year
! Title
! Role
! Notes
|-
|1938||''[[A Yank at Oxford]]''||Minor Role||Uncredited
|-
|1939||''[[The Four Just Men (1939 film)|The Four Just Men]]''||Rally campaigner||Uncredited
|-
|rowspan=3|1948||''[[Trouble in the Air]]''||Truelove||
|-
|''[[William Comes to Town]]''||Circus Superintendent||
|-
|''[[A Piece of Cake]]''||Mr. Short||
|-
|rowspan=4|1949||''[[Murder at the Windmill]]''||Detective Sergeant||(as Jon Pertwer)
|-
|''[[Helter Skelter (1949 film)|Helter Skelter]]''||Headwaiter / Charles II||
|-
|''[[Dear Mr. Prohack]]''||Plover||
|-
|''[[Miss Pilgrim's Progress]]''||Postman Perkins||
|-
|1950||''[[The Body Said No!]]''||Watchman||
|-
|1951||''[[Mister Drake's Duck]]''||Reuben||
|-
|1953||''[[Will Any Gentleman...?]]''||Charley Sterling||
|-
|1954||''[[The Gay Dog]]''||A Betting Man||
|-
|1955||''[[A Yank in Ermine]]''||Slowburn Jenks||
|-
|1956||''[[It's a Wonderful World (1956 film)|It's a Wonderful World]]''||Conductor||Uncredited
|-
|1959||''[[The Ugly Duckling (1959 film)|The Ugly Duckling]]''||Victor Jekyll||
|-
|rowspan=2|1960||''[[Just Joe]]''||Prendergast||
|-
|''[[Not a Hope in Hell]]''||Dan||
|-
|1961||''[[Nearly a Nasty Accident]]''||Gen. Birkinshaw||
|-
|1963||''[[Ladies Who Do]]''||Sidney Tait||
|-
|1964||''[[Carry On Cleo]]''||Soothsayer||
|-
|rowspan=3|1965||''How to Undress in Public Without Undue Embarrassment''|| || Uncredited
|-
|''[[I've Gotta Horse]]''||Costumier's assistant||
|-
|''[[You Must Be Joking! (1965 film)|You Must Be Joking!]]''||Storekeeper (Hare Factory)||
|-
|1965||''[[Carry On Cowboy]]''||Sheriff Albert Earp||
|-
|rowspan=3|1966||''[[Carry On Screaming!]]''||Doctor Fettle||
|-
|''[[A Funny Thing Happened on the Way to the Forum (film)|A Funny Thing Happened on the Way to the Forum]]''||Crassus||
|-
|''[[Runaway Railway]]''||Station Master||
|- 
|rowspan=2|1969||''[[Up in the Air (1969 film)|Up in the Air]]''||Figworthy||
|-
|''Under the Table You Must Go''||Himself||Documentary
|-
|1970||''I Understand''||Uncredited||Short
|-
|1971||''[[The House That Dripped Blood]]''||Paul Henderson (segment 4 "The Cloak")||
|-
|1975||''[[One of Our Dinosaurs Is Missing]]''||Colonel||
|-
|rowspan=2|1977||''[[Adventures of a Private Eye]]''||Judd Blake||
|-
|''[[No. 1 of the Secret Service]]''||The Rev. Walter Braithwaite||
|-
|rowspan=2|1978||''[[The Water Babies (film)|The Water Babies]]''||Salmon / Kraken||(voice)
|-
|''[[Wombling Free]]''||Womble||(voice)
|-
|1982||''[[The Boys in Blue]]''||Coastguard||
|-
|1992||''[[Carry On Columbus]]''||Duke of Costa Brava||
|-
|1994||''[[Cloud Cuckoo]]''||Grandfather||Short
|-
|}"""

def WikiTable2DataFrame(table):
    list_of_lists = []

    a= table.split("\n|-")
    categories = a[1]
    categories = categories.split("\n! ")[1:] 
    
    rowString = "rowspan"
    span = 0
    savedIndex = -1;
    savedValue = "";
    
    for i in range(2,len(a)-1):
        b=a[i].split("\n|")[1]
        c= b.split("||")
        if span>0:
            c.insert(savedIndex,savedValue)
            span=span-1     
        for j,s in enumerate(c):
            if rowString in s:
                tab = s.split("|")
                savedIndex = j
                savedValue = tab[1]
                span = int(tab[0].split("=")[1])-1
                c[j]=savedValue
        list_of_lists.append(c)
        
    return pd.DataFrame(list_of_lists, columns=categories)
        
WikiTable2DataFrame(table)

Year                                              Title  \
0   1938                           ''[[A Yank at Oxford]]''   
1   1939  ''[[The Four Just Men (1939 film)|The Four Jus...   
2   1948                         ''[[Trouble in the Air]]''   
3   1948                      ''[[William Comes to Town]]''   
4   1948                            ''[[A Piece of Cake]]''   
5   1949                     ''[[Murder at the Windmill]]''   
6   1949  ''[[Helter Skelter (1949 film)|Helter Skelter]]''   
7   1949                           ''[[Dear Mr. Prohack]]''   
8   1949                    ''[[Miss Pilgrim's Progress]]''   
9   1950                          ''[[The Body Said No!]]''   
10  1951                        ''[[Mister Drake's Duck]]''   
11  1953                     ''[[Will Any Gentleman...?]]''   
12  1954                                ''[[The Gay Dog]]''   
13  1955                           ''[[A Yank in Ermine]]''   
14  1956  ''[[It's a Wonderful World (1956 film)|It's a ...   
15  1959  ''[[The Ugly Duckling (1959 film)|The Ugly Duc...   
16  1960                                   ''[[Just Joe]]''   
17  1960                         ''[[Not a Hope in Hell]]''   
18  1961                    ''[[Nearly a Nasty Accident]]''   
19  1963                              ''[[Ladies Who Do]]''   
20  1964                              ''[[Carry On Cleo]]''   
21  1965  ''How to Undress in Public Without Undue Embar...   
22  1965                           ''[[I've Gotta Horse]]''   
23  1965  ''[[You Must Be Joking! (1965 film)|You Must B...   
24  1965                            ''[[Carry On Cowboy]]''   
25  1966                        ''[[Carry On Screaming!]]''   
26  1966  ''[[A Funny Thing Happened on the Way to the F...   
27  1966                            ''[[Runaway Railway]]''   
28  1969    ''[[Up in the Air (1969 film)|Up in the Air]]''   
29  1969                    ''Under the Table You Must Go''   
30  1970                                   ''I Understand''   
31  1971               ''[[The House That Dripped Blood]]''   
32  1975            ''[[One of Our Dinosaurs Is Missing]]''   
33  1977                ''[[Adventures of a Private Eye]]''   
34  1977                ''[[No. 1 of the Secret Service]]''   
35  1978   ''[[The Water Babies (film)|The Water Babies]]''   
36  1978                              ''[[Wombling Free]]''   
37  1982                           ''[[The Boys in Blue]]''   
38  1992                          ''[[Carry On Columbus]]''   
39  1994                               ''[[Cloud Cuckoo]]''   

                                      Role             Notes  
0                               Minor Role        Uncredited  
1                         Rally campaigner        Uncredited  
2                                 Truelove                    
3                    Circus Superintendent                    
4                                Mr. Short                    
5                       Detective Sergeant  (as Jon Pertwer)  
6                  Headwaiter / Charles II                    
7                                   Plover                    
8                          Postman Perkins                    
9                                 Watchman                    
10                                  Reuben                    
11                        Charley Sterling                    
12                           A Betting Man                    
13                          Slowburn Jenks                    
14                               Conductor        Uncredited  
15                           Victor Jekyll                    
16                             Prendergast                    
17                                     Dan                    
18                         Gen. Birkinshaw                    
19                             Sidney Tait                    
20                              Soothsayer                    
21                                

In [64]:
df_actors.head(5)

Death Date  \
Name                  Birth Date               
Jack Weston           1924        1996-05-03   
John Beradino         1917        1996-05-19   
Jon Pertwee           1919        1996-05-20   
Lash LaRue            1917        1996-05-21   
Enrique Álvarez Félix 1934        1996-05-24   

                                                             Description  \
Name                  Birth Date                                           
Jack Weston           1924                             , american actor    
John Beradino         1917         , american baseball player and actor    
Jon Pertwee           1919                              , british actor    
Lash LaRue            1917                             , american actor    
Enrique Álvarez Félix 1934                              , mexican actor    

                                  Actor  Author  Musician  
Name                  Birth Date                           
Jack Weston           1924         True   False     False  
John Beradino         1917         True   False     False  
Jon Pertwee           1919         True   False     False  
Lash LaRue            1917         True   False     False  
Enrique Álvarez Félix 1934         True   False     False